### DISCUSSION TITLE
Find users outside of Free Pascal?

### DISCUSSION THREAD

**Discussion by ghost** [fredvs/ideU#64] [Q&A] [answered]
It seems MSEGUI + FPGUI don't have much interest from the Free Pascal users! They are so accustomed to the LCL abstraction of Lazarus that the only way to make them use MSEGUI + FPGUI is having a MSEGUI + FPGUI widgetset for the LCL! A widgetset for LCL based on FPGUI is already available but is no longer developed. A widgetset for LCL based on MSEGUI is simply nonexistent. I think MSEGUI + FPGUI is good GUI library. Do you ever think about making bindings to use with other languages?

Since they are written using OOP Free Pascal so will have to be flatten to plain C API like qt5pas to be able to use with other languages (at least it's what I think).

**Comment by fredvs** [comment, marked as answer]
Hello and welcome to ideU.

Yes, indeed Lazarus has more success than MSEgui and fpGUI.

But once you have work with LCL and are tired by the many bugs of GTK, Qt and friends (that you dont have easy access to fix it) and have try a stable widget like MSEgui, you will think different.

About MSEgui for C of course it would be fantastic for C developers to have a wonderful widgetset like MSEgui.
But MSEgui has nearly 10000 commits and the work to translate all the Pascal code into C would be lot of lot of work.
Anyway, MSEgui is open source and every pull-request is welcome ;-)

**Reply by ghost** [reply]
If MSEGUI is too big, what about FPGUI?

**Comment by fredvs** [comment]
Hello.
With fpGUI there would be less work but even lot of work to translate all the code into C.
